atintegrators/BndStrMPoleSymplectic4RadPass

```diff
@@ -0,0 +1,429 @@
+        int MaxOrder, NumIntSteps;
+        double *PolynomA, *PolynomB, *R1, *R2, *T1, *T2;
+        double *r_in;
+        const mxArray *ElemData = prhs[0];
+        int num_particles = mxGetN(prhs[1]);
+        Length=atGetDouble(ElemData,"Length"); check_error();
+        BendingAngle=atGetDouble(ElemData,"BendingAngle"); check_error();
+        EntranceAngle=atGetDouble(ElemData,"EntranceAngle"); check_error();
+        ExitAngle=atGetDouble(ElemData,"ExitAngle"); check_error();
+        PolynomA=atGetDoubleArray(ElemData,"PolynomA"); check_error();
+        PolynomB=atGetDoubleArray(ElemData,"PolynomB"); check_error();
+        MaxOrder=atGetLong(ElemData,"MaxOrder"); check_error();
+        NumIntSteps=atGetLong(ElemData,"NumIntSteps"); check_error();
+        Energy=atGetDouble(ElemData,"Energy"); check_error();
+        /*optional fields*/
+        Gap=atGetOptionalDouble(ElemData,"FullGap", 0); check_error();
+        Fint1=atGetOptionalDouble(ElemData,"FringeInt1", 0); check_error();
+        Fint2=atGetOptionalDouble(ElemData,"FringeInt2", 0); check_error();
+        R1=atGetOptionalDoubleArray(ElemData,"R1"); check_error();
+        R2=atGetOptionalDoubleArray(ElemData,"R2"); check_error();
+        T1=atGetOptionalDoubleArray(ElemData,"T1"); check_error();
+        T2=atGetOptionalDoubleArray(ElemData,"T2"); check_error();
+        X0ref=atGetOptionalDouble(ElemData,"X0ref", 0); check_error();
+        ByError=atGetOptionalDouble(ElemData,"ByError", 0); check_error();
+        RefDZ=atGetOptionalDouble(ElemData,"RefDZ", 0); check_error();
+        irho = BendingAngle/Length;
+        flen = 2.0/irho*sin(BendingAngle/2.0); /* field length */
+        if (mxGetM(prhs[1]) != 6) mexErrMsgIdAndTxt("AT:WrongArg","Second argument must be a 6 x N matrix");
+        /* ALLOCATE memory for the output array of the same size as the input */
+        plhs[0] = mxDuplicateArray(prhs[1]);
+        r_in = mxGetDoubles(plhs[0]);
+        BndStrMPoleSymplectic4RadPass(r_in, flen, irho, PolynomA, PolynomB,
+            MaxOrder, NumIntSteps, EntranceAngle, ExitAngle, X0ref, ByError,
+            RefDZ, Fint1, Fint2, Gap, T1, T2, R1, R2, Energy, num_particles);
+    }
+    else if (nrhs == 0) {
+        /* list of required fields */
+	    plhs[0] = mxCreateCellMatrix(8,1);
+	    mxSetCell(plhs[0],0,mxCreateString("Length"));
+	    mxSetCell(plhs[0],1,mxCreateString("BendingAngle"));
+	    mxSetCell(plhs[0],2,mxCreateString("EntranceAngle"));
+	    mxSetCell(plhs[0],3,mxCreateString("ExitAngle"));
+        mxSetCell(plhs[0],4,mxCreateString("PolynomA"));
+	    mxSetCell(plhs[0],5,mxCreateString("PolynomB"));
+	    mxSetCell(plhs[0],6,mxCreateString("MaxOrder"));
+	    mxSetCell(plhs[0],7,mxCreateString("NumIntSteps"));
+        mxSetCell(plhs[0],8,mxCreateString("Energy"));
+	    if(nlhs>1) {    /* list of optional fields */
+	        plhs[1] = mxCreateCellMatrix(7,1);
```

Comment:
And this should be plhs[1] = mxCreateCellMatrix(10,1);

Reply:
Thank you for the catch!